### PR TITLE
test(core): add 100% unit test coverage for telescope-file-operations and telescope-socket-setup

### PR DESCRIPTION
**Pull request by piconkayal**
# Pull Request

## Description

Added 100% unit test coverage for **telescope-file-operations** and **telescope-socket-setup** files. This ensures comprehensive testing of the functionalities within these files.

## How Has This Been Tested?

Unit tests have been written for all code paths within **telescope-file-operations.test.ts** and **telescope-socket-setup.test.ts**. These tests cover various input scenarios and edge cases.

## Checklist:

Before submitting your PR, please review the following checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Additional context
image

### CODE REVIEW

File: packages/node-telescope/test/core/base/telescope-socket-setup.test.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    io = new SocketServer();
+    socket = createMockSocket();
+    storage = createMockStorage();
+
+    jest.spyOn(io, 'on').mockImplementation((event, callback) => {
+      if (event === 'connection') {
+        callback(socket);
+      }
+
+      // TODO: Provide valid return type for 'on' event
```
Comment (mdipanjan) [inline packages/node-telescope/test/core/base/telescope-socket-setup.test.ts:47]: @piconkayal remember to replace this with actual implementation later.

File: packages/node-telescope/test/core/base/telescope-socket-setup.test.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  it('should set up event listeners for socket events', () => {
+    expect(socket.on).toHaveBeenCalledWith(EventTypes.GET_INITIAL_ENTRIES, expect.any(Function));
+    expect(socket.on).toHaveBeenCalledWith(EventTypes.GET_ENTRY_DETAILS, expect.any(Function));
+    expect(socket.on).toHaveBeenCalledWith('disconnect', expect.any(Function));
+    expect(storage.on).toHaveBeenCalledWith(EventTypes.NEW_ENTRY, expect.any(Function));
+  });
+
+  it('should handle GET_INITIAL_ENTRIES correctly', async () => {
+    const params = { type: EntryType.REQUESTS, page: 1, perPage: 10 };
+    const mockEntries = {
```
Comment (mdipanjan) [inline packages/node-telescope/test/core/base/telescope-socket-setup.test.ts:71]: mocks are supposed to be outside and imported from mocks folder, imo.

**Review by mdipanjan — APPROVED** [review]
LGTM
